### ISSUE TITLE
rtorrent RPC issue

### ISSUE THREAD

**Issue by waz1500** [evilhero/mylar#1374]
Hi,
I've been struggling to get Mylar to connect to rtorrent via RPC. I keep getting nothing but 401 unauthorised errors in mylar's logs. I'm using the development version of Mylar and connecting to 32P via Auth Mode.

I've used trying to use:
http://SERVER_IP:80/RPC2    - 401 unauthorised
https://SERVER_IP:443/RPC2  - 401 unauthorised
http://SERVER_IP/sadpanda/RPC2  - 401 unauthorised
https://SERVER_IP/sadpanda/RPC2 - 401 unauthorised
http://SERVER_IP/rutorrent/plugins/httprpc/action.php   - 401 unauthorised
http://SERVER_IP/rutorrent/plugins/rpc/action.php - 401 unauthorised

I've checked to see if its a bug within the software by trying another server of mine which is running an nginx based install and can confirm that the specified format http://SERVER_IP:80/RPC2 does indeed work with another server IP of mine.

Now 401 unauthorised access would suggest incorrect credentials. I thought so too, and so checked Mylar's config file and can confirm that the user/pass to access my rutorrent are indeed correct with no mistakes or typos. I can also rule that out because all I have to do is replace the server ip with my old server's IP and and the connection is successful.

I also get non-implemented errors when I try to use something like
scgi://127.0.0.1:45228 or localhost:45228

UPDATE**
*I have posted this on the Quickbox forums to try to find an answer and they mentioned that I would need digest authorisation enabled for the communication to be successful.
I determined that my nginx server uses basic authorisation whereas the second server uses digest authorisation. Is there a way to set to set the authentication from basic authorisation to digest authorisation in the config.ini?
Something like torrent_auth_type = digest like SickRage uses in it's own config.ini

UPDATE2**
I managed to get this working by using the scgi://127.0.0.1:45228 without the credentials. I never realised that leaving the user pass filled in would cause it to give non-implemented errors in mylar's logs.

**Comment by evilhero** [comment]
Have you tried RPC1 and/or confirmed that the RPC plugin is loaded properly within rutorrent? Depending on where you have your box, it might depend on the provider (some don't open tbe RPC port, but allow the httprpc module in lieu of) 

I don't think the version of rtorrent would matter, but are the versions different between the servers?

**Comment by waz1500** [comment]
Thanks for responding evilhero. Luckily I managed to find the issue before your response. I assumed that Mylar would support digest authentication and so no wonder I kept getting 401 errors even though the credentials were correct. It could be worth putting somewhere that currently Mylar only supports Basic Authentication.  

It could be also be worth putting a little more info on possible mounts under rtorrent Host:Port. Currently you have `ie. https://my.rtorrent:80/myuser/RPC1`" but could also put something like [EXAMPLE](http://i.imgur.com/LydymvB.jpg) to show other possible mounts that Mylar supports.

If you do get Digest Auth mode working, I think it would be a great addition :)

**Comment by evilhero** [comment]
Just an update - I ended up getting a new box which only used digest authentication (I also use quickbox, but was using an old version that I couldn't update until now), so I've been working on this for the last few hours. 

Got digest authentication working in a very simple way, but at least it's working using both the httprpc plugin and the rpc plugin (verified against http://SERVER_IP/user/RPC2 and http://SERVER_IP/rutorrent/plugins/httprpc/action.php).

Hope to get this out in the next day or so into the development branch ;)

**Comment by evilhero** [comment]
Sorry about the delay, personal life and then the shopping extravaganza this weekend -  been crazy. Plus I ended up trashing my rtorrent modules during this process, so I had to start from the initial base again. Looks like it's working now though, and here's an updated screenshot of the new config just so you know I haven't forgotten :)
 
![rtorrent_options](https://cloud.githubusercontent.com/assets/909424/20682535/6c33a010-b576-11e6-898e-39ce5325d713.png)

Hope to get it out fairly soon, there's some things that are intertwined in these modules so that's why it's taking longer to get it all out.
